pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -156,15 +156,16 @@ func NewController(podInformer coreinformers.PodInformer,
 		c.topologyCache = topologycache.NewTopologyCache()
 	}
 
-	c.reconciler = &reconciler{
-		client:               c.client,
-		nodeLister:           c.nodeLister,
-		maxEndpointsPerSlice: c.maxEndpointsPerSlice,
-		endpointSliceTracker: c.endpointSliceTracker,
-		metricsCache:         endpointslicemetrics.NewCache(maxEndpointsPerSlice),
-		topologyCache:        c.topologyCache,
-		eventRecorder:        c.eventRecorder,
-	}
+	c.reconciler = endpointslicerec.NewReconciler(
+		c.client,
+		c.nodeLister,
+		c.maxEndpointsPerSlice,
+		c.endpointSliceTracker,
+		endpointslicemetrics.NewCache(maxEndpointsPerSlice),
```

Comment:
Can `NewCache` be called inside `NewReconciler` instead by taking in `maxEndpointsPerSlice` as an argument? This would eliminate `endpointslicemetrics` as a direct dependency.

Reply:
It will not remove the dependency, it is still used to handle the metrics.

Reply:
It will be removed as a direct dependency and as an import.